### ISSUE TITLE
When using Live Share onFocusChange doesn't trigger Auto Save

### ISSUE THREAD

**Issue by leandrotoledo** [microsoft/vscode#122793]
Steps to reproduce:

1. Invite someone to code Live Share with Read/Write permissions
2. When the guest edits a file, and then proceeds to click away (e.g. Terminal)
3. File should be saved, but it's not getting saved.

Both host and guest have Files: Auto Save with onFocusChange in the local VS Code settings.

CPUs | AMD Ryzen 5 5600X 6-Core Processor (12 x 4199)
-- | --
GPU Status | 2d_canvas: unavailable_software gpu_compositing: disabled_software multiple_raster_threads: enabled_on oop_rasterization: disabled_off opengl: disabled_off protected_video_decode: disabled_off rasterization: disabled_software skia_renderer: enabled_on video_decode: disabled_software vulkan: disabled_off webgl: unavailable_software webgl2: unavailable_software
Load (avg) | 1, 2, 1
Memory (System) | 31.33GB (21.93GB free)
Process Argv | --no-sandbox --unity-launch --crash-reporter-id c677f2fe-9ceb-40ac-a83e-6ad993316d25
Screen Reader | no
VM | 0%

Extension | Author (truncated) | Version
-- | -- | --
python | ms- | 2021.4.765268190
jupyter | ms- | 2021.5.745244803
cpptools | ms- | 1.3.1
sublime-keybindings | ms- | 4.0.9
vsliveshare | ms- | 1.0.4131
vscode-arduino | vsc | 0.4.2

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by leandrotoledo** [comment]
Sure. The assumption here is that both host and guests will be running the VS Code Insider Build with `onFocusChange` enabled.

Steps followed -

### Setting the stage
1. Downloaded `https://code.visualstudio.com/sha/download?build=insider&os=linux-x64`
2. `$ tar -zxvf code-insider-x64-1619825818.tar.gz`
3. `$ cd VSCode-linux-x64/`
4. `./code-insiders`
5. On Settings, changed Files: Auto Save to `onFocusChange`
6. On Extension, installed Visual Studio Live Share
7. Authenticated via Github, I created a new Live Share session and sent the link to a colleague using VS Code
8. Guest accepted the invitation. Host accepted Guest with `Read/Write` permissions, including a `bash (Read/Write) Shared Terminal`.

### Host triggering onFocusChange (Working as Intended)
9. Host (running VS Code Insider `1.56.0-insider`) created `test` file. Host starts typing and the small dot on the file's tab indicates the file is not saved. 
10. Host clicked on Terminal, run `$ cat test` and confirms contents have been automatically saved.

### Guest triggering onFocusChange (Bug)
9. Host (running VS Code Insider `1.56.0-insider`) updates `test` file.
10. Guest clicked on Terminal, run `$ cat test` and confirms contents **have not** been saved.
11. Host confirms that the small dot on the file's tab is present still.

@leandromuto will attach a screen capture as he was the guest in this exercise.

**Comment by bpasero** [comment]
A guest cannot trigger the save on the host, I don't think this is possible without some APIs for Live Share to forward a save to the host.

**Comment by bpasero** [comment]
I cannot reproduce, but it required me on the "Guest" to configure `"files.autoSave": "onFocusChange"` as well. In other words, the settings of the host are not applied to the guest.

**Comment by bpasero** [comment]
![recording](https://user-images.githubusercontent.com/900690/116820544-07c87380-ab76-11eb-8b1c-97f9630981d1.gif)

**Comment by leandromuto** [comment]
Hi @bpasero 

Thanks for make it clear. You can see the evidence in the following video https://drive.google.com/file/d/1GpeemrU9QqV3zI2rpzd0qUfY9rIV8EDv/view?usp=sharing

**Comment by bpasero** [comment]
As you can see from my "evidence", I cannot reproduce so there are more steps needed. Here is what I did for you to reproduce:
* use insiders
* run `code-insiders --user-data-dir <empty folder> --extensions-dir <empty folder>
* install LiveShare
* run `code-insiders --user-data-dir <other empty folder> --extensions-dir <other empty folder>
* install LiveShare
* in the first instance open a folder and share it (read/write)
* join from the second instance
* configure auto save on focus change in both windows
* in both instances make a file dirty and move focus to another part of the UI
* in both cases the file saves for me

**Comment by leandrotoledo** [comment]
Hi Ben, from your screen capture it seems you had two instances of VS Code
running on the same computer/session? Could it be that? The step by step
you posted is exactly what @leandromuto and I have done but 10,000 miles
from each other :-)

I'm happy to give it another try anyways.

Thanks!

**Comment by bpasero** [comment]
Yeah but that should really not make any difference, the 2 instances having separate user data directories should simulate 2 distinct sessions.
